mysite/settings/production.py

```diff
@@ -1,7 +1,79 @@
+import os
+import random
+import string
 from .base import *
 
 DEBUG = False
 
+# DJANGO_SECRET_KEY *should* be specified in the environment. If it's not, generate an ephemeral key.
+if "DJANGO_SECRET_KEY" in os.environ:
+    SECRET_KEY = os.environ["DJANGO_SECRET_KEY"]
+else:
+    # Use if/else to avoid calculating this if we don't need it.
+    print(
+        "WARNING: DJANGO_SECRET_KEY missing. Generating ephemeral SECRET_KEY."
+    )
+    SECRET_KEY = "".join(
+        [random.SystemRandom().choice(string.printable) for i in range(50)]
+    )
+
+
+# Make sure Django can detect a secure connection properly on Heroku:
+SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
+
+# Force HTTPS redirect (enabled by default!)
+# https://docs.djangoproject.com/en/stable/ref/settings/#secure-ssl-redirect
+SECURE_SSL_REDIRECT = True
+
+# IMPORTANT: Set this to a real hostname when using this in production!
+# See https://docs.djangoproject.com/en/3.2/ref/settings/#allowed-hosts
+ALLOWED_HOSTS = os.getenv("DJANGO_ALLOWED_HOSTS", "*").split(",")
+
+CSRF_TRUSTED_ORIGINS = os.getenv("DJANGO_CSRF_TRUSTED_ORIGINS", "").split(",")
+
+# Use the console email backend as we don’t configure emails in this tutorial.
+EMAIL_BACKEND = "django.core.mail.backends.console.EmailBackend"
+
+# Simplified static file serving.
+# https://warehouse.python.org/project/whitenoise/
+MIDDLEWARE.append("whitenoise.middleware.WhiteNoiseMiddleware")
+STATICFILES_STORAGE = "whitenoise.storage.CompressedManifestStaticFilesStorage"
+
+if "AWS_STORAGE_BUCKET_NAME" in os.environ:
+    AWS_STORAGE_BUCKET_NAME = os.getenv("AWS_STORAGE_BUCKET_NAME")
+    AWS_S3_REGION_NAME = os.getenv("AWS_S3_REGION_NAME")
+    AWS_S3_ENDPOINT_URL = os.getenv("AWS_S3_ENDPOINT_URL")
+    AWS_S3_ACCESS_KEY_ID = os.getenv("AWS_S3_ACCESS_KEY_ID")
+    AWS_S3_SECRET_ACCESS_KEY = os.getenv("AWS_S3_SECRET_ACCESS_KEY")
```

Comment:
Mentions of `os.getenv()`, but no mention of environment variables (as far as I know). I'm guessing this will be taken care of in the deployment part of the docs, but wanted to mention it so we don't forget. Deployment is hard for a lot of people, and specific instructions make the world of a difference.

Reply:
Yup! I’m sure we’ll miss a few things but that one is already tracked as one of the concepts we’ll have to explain (see the list of all bakerydemo concepts we identified, and a diff compared to the initial "getting started" tutorial: https://gist.github.com/thibaudcolas/f1507030e5e263592ff3d10c80b00ffe).

Our rough workflow currently is:

1. I produce the project’s code step-by-step as if I was implementing it as a dev(-ish)
2. Then do a rough description of the step-by-step changes in the README, mostly focusing on code correctness, order of the steps
3. @activus-d and I review the steps together to discuss key concepts to cover and possible pitfalls
4. @activus-d goes through my steps to make sure they work.
5. And then authors the full tutorial content (https://github.com/activus-d/tutorial-series/tree/introduction-to-contact-page)
6. Meagen then reviews